day_2/project_of_the_day.py

```diff
@@ -1,33 +1,40 @@
 # https://replit.com/@appbrewery/tip-calculator-end
 
-GREETING = 'Welcome to the tip calculator!'
-ASK_TOTAL_BILL = 'What was the total bill?'
-TIP_PERCENTS = (10, 12, 15)
-ASK_TIP_PERCENT = 'How much tip would you like to give? {}, {}, or {}?'.format(*TIP_PERCENTS)
-ASK_NUMBER_OF_PEOPLE = 'How many people to split the bill?'
-TELL_THE_RESULT = 'Each person should pay: ${:.2f}'
-VALUE_ERROR_MESSAGE = 'What you entered is not a number.'
+class TipCalculator:
+    GREETING = 'Welcome to the tip calculator!'
+    ASK_TOTAL_BILL = 'What was the total bill?'
+    TIP_PERCENTS = (10, 12, 15)
+    ASK_TIP_PERCENT = 'How much tip would you like to give? {}, {}, or {}?'.format(*TIP_PERCENTS)
+    ASK_NUMBER_OF_PEOPLE = 'How many people to split the bill?'
+    TELL_THE_RESULT = 'Each person should pay: ${:.2f}'
+    VALUE_ERROR_MESSAGE = 'What you entered is not a number.'
 
-print(GREETING)
+    def run_calculator(self):
+        print(self.GREETING)
 
-while True:
-    try:
-        bill = float(input(ASK_TOTAL_BILL))
-        tip_percent = int(input(ASK_TIP_PERCENT))
-    except ValueError:
-        print(VALUE_ERROR_MESSAGE)
-        continue
-    if tip_percent not in TIP_PERCENTS:
-        print('Please, choose the tip amount from the given options.')
-        continue
-    try:
-        num_of_people = int(input(ASK_NUMBER_OF_PEOPLE))
-    except ValueError:
-        print('What you entered is not a number.')
-        continue
+        while True:
+            try:
+                bill = float(input(self.ASK_TOTAL_BILL))
+                tip_percent = int(input(self.ASK_TIP_PERCENT))
+            except ValueError:
+                print(self.VALUE_ERROR_MESSAGE)
+                continue
```

Comment:
Not exactly what I've meant :D
I was thinking about something more like that:
```python
def get_value(value_type, message):
    ...

bill = get_value(float, ASK_TOTAL_BILL)
tip_percent = get_value(int, ASK_TIP_PERCENT)
```

etc